org.eclipse.cdt.cross.arc.gnu/src/org/eclipse/cdt/cross/arc/gnu/common/StateChecker.java

```diff
@@ -96,12 +96,18 @@ private void warnUser(final IProject project,final String currentState, String p
             + "Your compiler plug-in' state is %s, but this project was created with a "
             + "compiler state %s.", currentState, projectState);
 
-    final boolean isHeadless = System.getProperty("eclipse.application")
-        .equals("org.eclipse.cdt.managedbuilder.core.headlessbuild");
-    if (isHeadless) {
+    final String applicationProperty = System.getProperty("eclipse.application");
```

Comment:
Please add a comment that describes the whole piece of code. "Try to use eclipse.application property to figure out if this is GUI or console run and to show user an error message in a proper way. Property may not be set, then it is unknown whether the GUI mode or headless is launched, therefore show warning both by creating marker 
and writing to the System.err."